lisa/tools/hyperv.py

```diff
@@ -293,6 +295,40 @@ def create_nat(self, name: str, ip_range: str) -> None:
             force_run=True,
         )
 
+    def add_nat_mapping(self, nat_name: str, internal_ip: str) -> int:
+        external_port = self._get_next_available_nat_port()
+
+        # delete existing NAT mapping
+        self.delete_nat_mapping(external_port)
```

Comment:
The unused ports should be deleted when VM is deleted too. Here should have a check if the port mapping exists, direct fails. It can help identifying potential leaking issues. If we found it's normal scenario, and then add this kind of prevention deletion.

Reply:
Added port mapping deletion at VM deletion and added a check at creation to abort if port already used.